pdr_backend/trueval/subgraph.py

```diff
@@ -0,0 +1,100 @@
+from typing import List, Optional
```

Comment:
It's confusing to have two subgraphy.py files, one in util/ and one in trueval/.

I recommend to either
- merge all the content of trueval/subgraph.py into util/subgraph.py, OR
- rename: trueval/subgraph.py -> util/subgraph_slots.py, util/subgraph.py -> util/subgraph_predict_contracts.py

And whichever you use, be sure to name the functions inside such that one can readily distinguish the query/get functionality of one, versus the other.

NOTE: I just renanamed predictContractss -> predictContracts in main, you'll want to merge that into your branch before changing filenames.